03 Writing Algorithms/03 Securities/99 Asset Classes/02 Equity Options/02 Handling Data/99 Examples.html

```diff
@@ -0,0 +1,200 @@
+<h4>Example 1: Get Mid Price For Individual Contracts</h4>
+<p>This example shows how to handle <code>QuoteBar</code> data for shortlisted Equity Option contracts to calculate mid price using bid close and ask close data, while <a href="/docs/v2/writing-algorithms/securities/asset-classes/equity-options/requesting-data/individual-contracts">filter individual option contracts and request data</a> using <code class="csharp">OptionChainProvider.GetOptionContractList</code><code class="python">self.option_chain_provider.get_option_contract_list</code> method for the contracts that expires within the current week. Using mid price, we can examine the market fair value of the Option and compare with model theoretical price.</p>
+<div class="section-example-container">
+	<pre class="csharp">public class EquityOptionHandlingDataAlgorithm : QCAlgorithm
+{
+    private Symbol _spy;
+    private List&lt;Symbol&gt; _contracts = new();
+        
+    public override void Initialize()
+    {
+        // Seed the price with last known price to ensure the underlying price data is available on initial option contract filtering.
+        SetSecurityInitializer((security) =&gt; new FuncSecuritySeeder(GetLastKnownPrices).SeedSecurity(security));
```

Comment:
We need to use the BrokerageModelSecurityInitializer in all example:
```csharp
SetSecurityInitializer(new BrokerageModelSecurityInitializer(BrokerageModel, new FuncSecuritySeeder(GetLastKnownPrices)));
```
Otherwise, we lose the brokerage model.